crates/ika-core/src/dwallet_mpc/network_dkg.rs

```diff
@@ -471,16 +454,69 @@ fn generate_ristretto_dkg_party_public_input(
     bcs::to_bytes(&public_params).map_err(|e| DwalletMPCError::BcsError(e))
 }
 
-pub(crate) fn dwallet_mpc_network_key_from_session_output(
+pub(crate) fn create_dwallet_mpc_network_decryption_key_from_onchain_public_output(
     epoch: u64,
     key_scheme: DWalletMPCNetworkKeyScheme,
     weighted_threshold_access_structure: &WeightedThresholdAccessStructure,
-    public_output: &[u8],
+    key_data: DWalletNetworkDecryptionKeyData,
+) -> DwalletMPCResult<NetworkDecryptionKeyShares> {
+    if (key_data.current_reconfiguration_public_output.is_empty()) {
+        create_dwallet_mpc_network_decryption_key_from_network_dkg_public_output(
+            epoch,
+            key_scheme,
+            weighted_threshold_access_structure,
+            &key_data.network_dkg_public_output,
+        )
+    } else {
+        create_dwallet_mpc_network_decryption_key_from_reshare_public_output(
+            epoch,
+            weighted_threshold_access_structure,
+            &key_data.current_reconfiguration_public_output,
+        )
+    }
+}
+
+fn create_dwallet_mpc_network_decryption_key_from_reshare_public_output(
+    epoch: u64,
+    weighted_threshold_access_structure: &WeightedThresholdAccessStructure,
+    public_output_bytes: &[u8],
+) -> DwalletMPCResult<NetworkDecryptionKeyShares> {
+    let public_output: <ReshareSecp256k1Party as mpc::Party>::PublicOutput =
+        bcs::from_bytes(&public_output_bytes)?;
+    let plaintext_space_public_parameters = group::PublicParameters::<secp256k1::Scalar>::default();
+    let encryption_scheme_public_parameters = public_output
+        .compute_encryption_scheme_public_parameters::<secp256k1::GroupElement>(
+            plaintext_space_public_parameters,
+            DEFAULT_COMPUTATIONAL_SECURITY_PARAMETER,
+        )
+        .map_err(|e| DwalletMPCError::ClassGroupsError(e.to_string()))?;
```

Comment:
also better to put it into the function

Reply:
but you don't need this at all, also don't need the field, you can just access `decryption_key_share_public_parameters.encryption_scheme_public_parameters`